ush/test/encodeBufr.sh

```diff
@@ -0,0 +1,215 @@
+    echo "Error: Invalid mode '$mode'. Expected 'bufr_backend' or 'script_backend' or 'bufr2netcdf' or 'script2netcdf'."
+    usage
+fi
+
+# =============================================
+# Validate nproc is a positive integer or zero
+# =============================================
+if ! [[ "$nproc" =~ ^[0-9]+$ && "$nproc" -ge 0 ]]; then
+	echo "Error: nproc must be a positive integer or zero."
+    usage
+fi
+
+# =========================
+# Check if all checks pass
+# =========================
+echo "root directory of obsForge: $obsforge_dir"
+echo "cycle: $cycle"
+echo "bufrtype: $bufrtype"
+echo "obstype: $obstype"
+echo "sensor: $sensor"
+echo "split_by_category: ${split_by_category}"
+echo "mode: $mode"
+echo "number of processors: $nproc"
+
+# ======================
+# Enable debugging mode
+# ======================
+set -x
+
+# =========================
+# Set unlimited stack size
+# =========================
+ulimit -s unlimited
+ulimit -a
+
+# ====================================
+# Set OOPS run time output parameters 
+# ====================================
+export OOPS_TRACE=1
+export OOPS_DEBUG=1
+
+# =======================================================
+# Set WXFLOW log level
+# Valid Log_LEVEL: DEBUG, INFO, WARNING, ERROR, CRITICAL
+# =======================================================
+export LOG_LEVEL=DEBUG
+
+# ===============================
+# Load obsForge required modules 
+# ===============================
+module use ${obsforge_dir}/modulefiles
+module load obsforge/hera.intel || { echo "Error loading obsforge module"; exit 1; }
+module list
+
+# ==============================
+# Set bufr-query python library
+# ==============================
+export LD_LIBRARY_PATH="${obsforge_dir}/build/lib:${LD_LIBRARY_PATH}"
+export PYTHONPATH="${PYTHONPATH}:${obsforge_dir}/build/lib/python3.10/site-packages"
+
+# ========================
+# Set ioda python library
+# =========================
+export PYTHONPATH="${PYTHONPATH}:${obsforge_dir}/build/lib/python3.10"
+
+# ============
+# Set wxfloww 
+# ============
+export PYTHONPATH="${PYTHONPATH}:${obsforge_dir}/sorc/wxflow/src"
+
+# ===========================
+# Configure SLUM environment
+# ===========================
+export SLURM_ACCOUNT=da-cpu
+export SALLOC_ACCOUNT=$SLURM_ACCOUNT
+export SBATCH_ACCOUNT=$SLURM_ACCOUNT
+export SLURM_QOS=debug
+
+# ===================================
+# Extract year, month, day, and hour
+# ===================================
+y4="${cycle:0:4}"
+m2="${cycle:4:2}"
+d2="${cycle:6:2}"
+h2="${cycle:8:2}"
+
+# ====================
+# Set directory paths
+# ====================
+work_dir="$PWD"
+out_dir="${work_dir}/testoutput/$cycle/${mode}"
+in_dir="${work_dir}/testinput/$cycle"
+mkdir -p -m770 ${out_dir} || { echo "Error creating output directory: ${out_dir}"; exit 1; }
+
+# ===============
+# Set file paths
+# ===============
+ioda_config_yaml="${work_dir}/bufr2ioda_${mode}_${obstype}.yaml"
```

Comment:
Suggest adding lines to move the appropriate files to the work_dir from their original locations. Written as is, the script looks for ioda_config_yaml, mapping_file, and input_file which are not already there.

Reply:
@CoryMartin-NOAA and @rmclaren Can we merge this branch?  
I have tested the files added thoroughly.